clients/src/main/java/org/apache/kafka/common/network/ChannelBuilder.java

```diff
@@ -32,10 +34,11 @@
      * returns a Channel with TransportLayer and Authenticator configured.
      * @param  id  channel id
      * @param  key SelectionKey
-     * @param  maxReceiveSize
+     * @param  maxReceiveSize max size of a single receive buffer to allocate
+     * @param  memoryPool memory pool from which to allocate buffers, or null for none
```

Comment:
I account for null as a safety net even though using NONE is clearer. so its both by design. having said that, i'll gladly go for one or the other if there's a style guideline.